### ISSUE TITLE
Can't connect to mysql instance after mysql_service install

### ISSUE THREAD

**Issue by daften** [chef-cookbooks/mysql#339]
This statement in a recipe
```
mysql_service 'default' do
  version '5.7'
  bind_address '0.0.0.0'
  port '3306'
  initial_root_password 'test'
  action [:create, :start]
end
```

And then using 
```
mysql -S /var/run/mysql-default/mysqld.sock -u root -p
```
with password test, does not allow me access
```
ERROR 1045 (28000): Access denied for user 'root'@'localhost' (using password: YES)
```

**Comment by daften** [comment]
Could this be because of line 153 in helpers.rb? 
```
GRANT ALL ON *.* TO 'root'@'%' WITH GRANT OPTION ;
```

Is that BTW not completely unsafe? Isn't it better to only allow root access from localhost?

**Comment by daften** [comment]
This is fixed in PR #341 .

**Comment by someara** [comment]
I think we need two separate rules... one for the TCP connection and one for the socket connection.

GRANT ALL ON *.* TO 'root'@'127.0.0.1' WITH GRANT OPTION ;
GRANT ALL ON *.* TO 'root'@'localhost' WITH GRANT OPTION ;

**Comment by someara** [comment]
Fixed up in v6.0.24
